packages/cli/src/commands/contract/compile.ts

```diff
@@ -25,82 +24,89 @@ export class CompileContract extends Command {
       default: false,
       char: "r",
       description: "A production contract should always be build in `release` mode for building optimized wasm"
+    }),
+    all: Flags.boolean({
+      default: false,
+      char: "a",
+      description: "Set all to true to compile all contracts"
     })
   };
 
   static args = [
     {
       name: "contractName",
-      required: true,
-      description: "Name of the contract to compile",
+      required: false,
+      default: "",
+      description: "Name of the contracts to compile, separated by comman for multiple contracts compilation",
     },
   ];
 
   async run(): Promise<void> {
     const { args, flags } = await this.parse(CompileContract);
 
-    await ensureSwankyProject();
+    if (args.contractName == "" && !flags.all) {
+      this.error("No contracts were selected to compile")
+    }
 
+    await ensureSwankyProject();
     const config = await getSwankyConfig();
 
-    const contractInfo = config.contracts[args.contractName];
-    if (!contractInfo) {
-      this.error(`Cannot find a contract named ${args.contractName} in swanky.config.json`);
+    const contractNames = [];
+    if (flags.all) {
+      const contractList = readdirSync(path.resolve("contracts"));
+      for (const contractName of contractList) {
+        if (lstatSync(path.resolve("contracts", contractName)).isDirectory()) {
```

Comment:
No need to call `lstat` on every file when you can get that info right away during `readdir` if you pass it `withFileTypes: true` option.